wolfcrypt/src/asn.c

```diff
@@ -20051,6 +20053,17 @@ int PemToDer(const unsigned char* buff, long longSz, int type,
                         }
                     }
                 #endif /* !NO_DES3 */
+                #if !defined(NO_AES) && defined(HAVE_AES_CBC) && \
+                    defined(HAVE_AES_DECRYPT)
+                    if (info->cipherType == WC_CIPHER_AES_CBC) {
+                        if (der->length > AES_BLOCK_SIZE) {
```

Comment:
What if `der->length` is a multiple of AES_BLOCK_SIZE? Will there still be padding? If not this logic isn't right. Perhaps the check should be `if ((der->length % AES_BLOCK_SIZE) != 0) {`?

Reply:
The check is to see whether there is enough bytes for at least one block - a padded block. Must have a full block of padding if the plaintext is a multiple of the block size - otherwise you don't know whether there is padding there or not.
The decryption will fail if the encrypted data isn't a multiple of the block size.